types/react-select/src/Select.d.ts

```diff
@@ -202,16 +205,17 @@ export interface NamedProps<OptionType extends OptionTypeBase = { label: string;
   /* Select the currently focused option when the user presses tab */
   tabSelectsValue?: boolean;
   /* The value of the select; reflected by the selected option */
-  value?: ValueType<OptionType>;
+  value?: ValueType<OptionType, IsMulti>;
 
   defaultInputValue?: string;
   defaultMenuIsOpen?: boolean;
-  defaultValue?: ValueType<OptionType>;
+  defaultValue?: IsMulti extends true ? ValueType<OptionType, boolean> : ValueType<OptionType, false>;
```

Comment:
This prevents arrays being passed in when in single mode, and allows both arrays or objects in multi